packages/docusaurus-theme-classic/src/types.d.ts

```diff
@@ -290,6 +290,7 @@ declare module '@theme/NavbarItem/DefaultNavbarItem' {
     activeClassName?: string;
     prependBaseUrlToHref?: string;
     isActive?: () => boolean;
+    position?: string;
```

Comment:
I  don't think this is an improvement. We shouldn't declare a useless props to filter it out deep in the tree. 
Instead we should ensure to never pass a position props to the NavLink in the first place.

Reply:
Yeah agree, apparently the problem lies here

https://github.com/facebook/docusaurus/blob/4130f1a1952bf2b359c36091a50d97db3454cf20/packages/docusaurus-theme-classic/src/validateThemeConfig.js#L48

But so far I don't really understand how to do this with Joi.

Reply:
Done in https://github.com/facebook/docusaurus/pull/3585/commits/d1b43cac85e400703a65981ea3ef38416a28a80f
But not sure if this is correct.

Reply:
> 9:47:03 PM: Error: "navbar.items[4].items[0]" contains link reference "ref:local:baseNavbarItem" which is outside of schema boundaries

Hm, locally it works for me...

Reply:
Fixed in https://github.com/facebook/docusaurus/pull/3585/commits/5b34e759181362bf13955711ba578e1d1622d3ff
I don't know for sure it's all correct (ha ha).

Reply:
haha, Joi is a bit annoying to understand for me too, the doc is quite painful to read. Start to regret a bit to not have chosen Yup (less powerful but simpler in practice imho).

@lex111 as this Joi thing is not obvious (for me too) what about adding this in a separate PR + adding validation tests? So that we can merge this PR first and find a good validation schema separately?

Reply:
I'm not very confident merging this schema update without tests 🤪 

In the past we had some issues on some releases because we had too strict validation rules, and some users wanted to pass additional attributes like target=_blank , data attributes or other fancy props that a link should accept, so we should rather ensure it's still possible after this refactor.

Reply:
> In the past we had some issues on some releases because we had too strict validation rules, and some users wanted to pass additional attributes like target=_blank , data attributes or other fancy props that a link should accept, so we should rather ensure it's still possible after this refactor.

I understand your concern, but after that change everything seems to work as before. The GitHub link has a custom attr ʻaria-label`. I just created a base type for the link, which then extends by the navbar item, and is also used for dropdown (child) links.

Alright, I'll try to write tests. Could you please help me describe which navbar cases need to be tested? It looks like we don't have enough tests for validation navbar.

Reply:
Basically we should additional item props to the Link, so that user can pass whatever he wants a link support (target, data attributes,...). So in the end you add objectSchema.unknown() and ensure that passing such attributes are accepted.

I don't find a good issue for those nav items validation regarding custom attributes, but here are various bug reports regarding validation that we had:
- https://github.com/facebook/docusaurus/issues/3349
- https://github.com/facebook/docusaurus/pull/3120

Reply:
> I don't think we can easily validate all additional props that a Link can accept, so not sure there's a better option than unknown()

I think using `unknown()` is a great option, the gist of my change is that we are not setting a default value for `position` and hence don't pass this prop to the component. I don’t understand yet what could be broken, but will write more tests.

Reply:
Tried testing all navbar items use cases in https://github.com/facebook/docusaurus/pull/3585/commits/4e1f5b716964e2647241c3194abe7cc2cfca76ea
Do you think it's right?

Reply:
that looks safe thanks, let's merge